accepted/2025/cli-acquisition-tool.md

```diff
@@ -0,0 +1,224 @@
+
+## Stakeholders and Reviewers
+
+- **.NET SDK Team**: Core implementation and CLI design
+- **VS Code C# Extension Team**: Integration scenarios and user experience
+- **Visual Studio Team**: Ensuring compatibility with existing VS workflows
+- **Security Team**: Signature verification and SDL compliance
+- **.NET Foundation**: Community feedback and adoption strategy
+- **Documentation Team**: Getting started guides and migration documentation
+
+## Design
+
+### Bootstrapper CLI Design
+
+The tool will follow the CLI design outlined in the [related PR](https://github.com/dotnet/designs/pull/335) and will include commands like
+
+* install specific versions of the SDK tooling (10.0.100)
+* install logically-named versions of the SDK tooling (10, LTS, etc)
+* list the installed SDKs
+* uninstall specific versions of the SDK tooling
+* install the versions of components required by a given repo (as specified by global.json)
+* provide update checks/out-of-date information to end users about the installs they have
+
+### Initial setup
+
+On first-run, `dnup-init` will configure the users' environment to support user-local hives.
+* `DOTNET_ROOT` will be set to point to the `dnup` user-local root,
+* `PATH` will have the `dnup` user-local root appended to it,
+* on Windows `dnup-init` will request to elevate to set a Windows Registry key that will prevent global .NET installers from setting the global .NET Install location on the System `PATH` (which would always overwrite the user `PATH`).
+  * this behavior _may_ be optional - we may choose to only run it by default if the user has a global .NET install already
+
+This configuration will be necessity be shell-specific, so the tool will need to detect the user's shell and modify the appropriate profile file (e.g. `.bashrc`, `.zshrc`, etc.) to set the `DOTNET_ROOT` and `PATH` variables.
+We have some prior art in the `dotnet` CLI codebase for detecting the shell that is in use, which can be reused here.
+The different shells (and to some extent different OS platforms) have different ways of distributing such changes, so the tool will need to know about those to be a bit opinionated about how it makes these changes.
+
+This shell-specific configuration will need to be independently run via a command, because the user may choose a different shell after the initial setup.
+
+### Install locations
+
+These installs will be located at a default _user-local hive_ instead of a globally-reachable system hive.
+This will allow users to manage new tooling versions without needing to elevate.
+The tool will support being configured to install to other locations via configuration file, so that simple use of the tool remains consistent.
+
+### Install provenance
+
+The data about installs will be sourced from the release manifests that are published on a regular cadence by the .NET Releases team.
```

Comment:
Is there a need for this to be configurable so that the tool can point to a mirror/private copy of the release manifest for some enterprise scenarions?